### PR TITLE
Feature absmax reduce support

### PR DESCRIPTION
**Pull request by egeor**
This PR introduces the absmax reduce rows/cols TPPs (e.g. needed during quantization).

### CODE REVIEW

File: src/generator_mateltwise_unary_binary_avx_avx512.c [hunk 1]
```diff
@@ -303,7 +303,14 @@ void libxsmm_generator_configure_avx512_vlens(const libxsmm_meltw_descriptor* i_
   if ( LIBXSMM_DATATYPE_IMPLICIT == libxsmm_meltw_getenum_precision(i_mateltwise_desc, LIBXSMM_MELTW_FIELD_OUT)) {
     i_micro_kernel_config->vlen_out = i_micro_kernel_config->vlen_in;
   } else  if ( LIBXSMM_DATATYPE_I8 == libxsmm_meltw_getenum_precision(i_mateltwise_desc, LIBXSMM_MELTW_FIELD_OUT)) {
-    i_micro_kernel_config->vlen_out = 64;
+    /* if the computation is done in F32 or the input is in F32, then set vlen_out to 16 */
```
Comment (alheinecke) [inline src/generator_mateltwise_unary_binary_avx_avx512.c:306]: Is this related to this requested change?

File: src/generator_matequation_regblocks_avx_avx512.c [hunk 2]
```diff
@@ -632,6 +632,10 @@ void libxsmm_generator_mateqn_load_arg_to_2d_reg_block( libxsmm_generated_code*
             char cvt_vname = (io_generated_code->arch < LIBXSMM_X86_AVX512_SKX) ? 'y' : 'z';
             libxsmm_generator_vcvthf8_to_f32_avx512_preppedstack( io_generated_code, cvt_vname,
                 cur_vreg, cur_vreg, cvt_vreg_aux0, cvt_vreg_aux1, cvt_mask_aux0, cvt_mask_aux1 );
+          } else if ( arg_info[i_arg_id].dtype == LIBXSMM_DATATYPE_I32 ) {
```
Comment (alheinecke) [inline src/generator_matequation_regblocks_avx_avx512.c:635]: Is this related? Also it seems we are doing the same thing 4 times, close at the same position in the code, could this be refactored?

**Review by alheinecke — COMMENTED** [review]
Minor questions.

File: src/generator_mateltwise_unary_binary_avx_avx512.c (hunk 1 above)
Comment (egeor) [reply src/generator_mateltwise_unary_binary_avx_avx512.c:306]: Yeah, this is was a bug pertaining to the quant unary TPP so technically not part of this PR (thus the first commit has these changes, but didn't want to open a PR just for this)

File: src/generator_matequation_regblocks_avx_avx512.c (hunk 2 above)
Comment (egeor) [reply src/generator_matequation_regblocks_avx_avx512.c:635]: Ditto as above, thus is handled in the first commit of this PR.

**Review by alheinecke — APPROVED** [review]
